app/src/main/kotlin/com/hippo/ehviewer/ui/screen/SearchBarScreen.kt

```diff
@@ -239,51 +243,66 @@ fun SearchBarScreen(
                     .padding(top = 48.dp) then scrollAwayModifier,
             )
         }
+        val onActiveChange = { v: Boolean ->
+            if (v) {
+                onSearchViewExpanded()
+            } else {
+                onSearchViewHidden()
+            }
+            active = v
+        }
+        val activeState = rememberCompositionActiveState()
         SearchBar(
-            modifier = Modifier.fillMaxWidth().align(Alignment.TopCenter) then scrollAwayModifier
+            modifier = Modifier.align(Alignment.TopCenter) then scrollAwayModifier
                 .windowInsetsPadding(WindowInsets.navigationBars.only(WindowInsetsSides.Horizontal)),
-            state = searchFieldState,
-            onSearch = {
-                hideSearchView()
-                onApplySearch()
-            },
-            active = active,
-            onActiveChange = {
-                if (it) {
-                    onSearchViewExpanded()
-                } else {
-                    onSearchViewHidden()
-                }
-                active = it
-            },
-            title = title.ifNotNullThen { Text(title!!, overflow = TextOverflow.Ellipsis) },
-            placeholder = searchFieldHint.ifNotNullThen { Text(searchFieldHint!!) },
-            leadingIcon = {
-                if (active) {
-                    IconButton(onClick = { hideSearchView() }) {
-                        Icon(Icons.AutoMirrored.Default.ArrowBack, contentDescription = null)
-                    }
-                } else {
-                    val drawerState = LocalNavDrawerState.current
-                    IconButton(onClick = { scope.launchUI { drawerState.open() } }) {
-                        Icon(Icons.Default.Menu, contentDescription = null)
-                    }
-                }
-            },
-            trailingIcon = {
-                if (active) {
-                    if (searchFieldState.text.isNotEmpty()) {
-                        IconButton(onClick = { searchFieldState.clearText() }) {
-                            Icon(Icons.Default.Close, contentDescription = null)
+            inputField = {
+                SearchBarInputField(
+                    state = searchFieldState,
+                    onSearch = {
+                        hideSearchView()
+                        onApplySearch()
+                    },
+                    expanded = active,
+                    onExpandedChange = onActiveChange,
+                    modifier = Modifier.widthIn(max = maxWidth - SearchBarHorizontalPadding * 2).fillMaxWidth(),
```

Comment:
`fillMaxWidth` should be applied after the `sizeIn` modifier in `SearchBarInputField`, otherwise that constraints will be obeyed.

Reply:
Done